### ISSUE TITLE
Conveniences ought to have a home

### ISSUE THREAD

**Issue by robrix** [antitypical/Result#124]
We’ve had a few PRs along the lines of #123, where someone has submitted a scratch for a not-uncommon itch. We’ve been turning them down—mostly—thus far, but maybe we should instead be finding them a better home.

As I understand it, this is the sort of thing that `git`’s [`contrib` dir](https://git.kernel.org/cgit/git/git.git/tree/contrib/README) exists to enable. Maybe we should consider something similar. Some ideas:

- a `Contrib` (sub?)framework encapsulating conveniences and experiments
- a section of the readme linking to gists with these things
- just including them in the framework and being done with it

**Comment by NachoSoto** [comment]
I like the idea! Especially as a separate framework.

**Comment by gfontenot** [comment]
I love the idea of linking to gists or providing a `Contrib` framework alongside `Result`. I don't think we should be including them in the main framework though.

**Comment by neilpa** [comment]
The one problem with a `Contrib` style framework in the same repo is it doesn't play all that nice with `carthage`.

**Comment by NachoSoto** [comment]
I was thinking it'd be a different repo for that reason.

**Comment by Thomvis** [comment]
Agreed, a separate framework & repo seems the best approach. In my experience however it is easy for the two frameworks to get out of sync, e.g. If you forget to update the contrib after a PR in the main repo. Is there a proven process for this?

Just my 5 cents for the name: `ResultOriented.framework`

**Comment by NachoSoto** [comment]
> If you forget to update the contrib after a PR in the main repo. Is there a proven process for this?

Well the "contrib" version would reference a specific `Result` version so compatibility would always be explicit.

That also means that one can simply change their `Carthage` reference from `Result` to this other framework, and their existing usages of `Result` would continue to work.

**Comment by Thomvis** [comment]
> Well the "contrib" version would reference a specific Result version so compatibility would always be explicit.

Understood. I was hoping there was some sort of safeguard/check that would prevent the contrib library from lagging behind.

**Comment by robrix** [comment]
@Thomvis: We could add it as a private dependency and then test it.

**Comment by kylef** [comment]
> In my experience however it is easy for the two frameworks to get out of sync, e.g. If you forget to update the contrib after a PR in the main repo.

Since Result just hit 1.0, I'd expect the API to be pretty stable and at least no backwards incompatible changes anytime soon. So I wouldn't expect this to be a problem.

**Comment by robrix** [comment]
Can we provide the contrib stuff in a submodule of the Result module? e.g.:

```swift
import Result.Oriented

let 🎉 = …
```

If so, we could ship it in this repo, which would be a lot more convenient both for contributors and end users.

**Comment by Evertt** [comment]
So uh, did this ever turn into something tangible?

**Comment by robrix** [comment]
@Evertt: Not thus far.

**Comment by Evertt** [comment]
Too bad, I really liked the idea. I happen to be a great fan of automagic stuff and since I know it won't be implemented in this library I'd love to know there's a central place where I can find plugins like that.

**Comment by robrix** [comment]
@Evertt: Nothing’s been established yet, but that’s not to say that you couldn’t be the one to establish it, if you want. If you’re interested, I’d suggest starting _very_ minimally, so that the contrib stuff’s home can be reviewed independently of any conveniences added therein.
